src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitResultContext.java

```diff
@@ -40,36 +43,51 @@ private static String getNonNullHeader(MessageHeaders headers, String name) {
         return header;
     }
 
-    public static VoltageInitResultContext fromMessage(Message<String> message) {
+    public static VoltageInitResultContext fromMessage(Message<String> message, ObjectMapper objectMapper) {
         Objects.requireNonNull(message);
         MessageHeaders headers = message.getHeaders();
-        UUID resultUuid = UUID.fromString(getNonNullHeader(headers, "resultUuid"));
-        UUID networkUuid = UUID.fromString(getNonNullHeader(headers, "networkUuid"));
+        UUID resultUuid = UUID.fromString(getNonNullHeader(headers, RESULT_UUID_HEADER));
+        UUID networkUuid = UUID.fromString(getNonNullHeader(headers, NETWORK_UUID_HEADER));
         String variantId = (String) headers.get(VARIANT_ID_HEADER);
         String receiver = (String) headers.get(HEADER_RECEIVER);
         String userId = (String) headers.get(HEADER_USER_ID);
-        Map<String, Double> voltageLevelsIdsRestricted = (Map<String, Double>) headers.get(VOLTAGE_LEVELS_IDS_RESTRICTED);
-
-        UUID parametersUuid = headers.containsKey(PARAMETERS_UUID_HEADER) ? UUID.fromString((String) headers.get(PARAMETERS_UUID_HEADER)) : null;
-        UUID reportUuid = headers.containsKey(REPORT_UUID_HEADER) ? UUID.fromString((String) headers.get(REPORT_UUID_HEADER)) : null;
+        Map<String, Double> voltageLevelsIdsRestricted = new HashMap<>();
+        if (headers.containsKey(VOLTAGE_LEVELS_IDS_RESTRICTED) && headers.get(VOLTAGE_LEVELS_IDS_RESTRICTED) != null) {
+            try {
+                TypeReference<Map<String, Double>> typeRef = new TypeReference<>() { };
+                voltageLevelsIdsRestricted = objectMapper.readValue((String) headers.get(VOLTAGE_LEVELS_IDS_RESTRICTED), typeRef);
+            } catch (JsonProcessingException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        UUID parametersUuid = null;
+        if (headers.containsKey(PARAMETERS_UUID_HEADER)) {
+            parametersUuid = UUID.fromString(getNonNullHeader(headers, PARAMETERS_UUID_HEADER));
+        }
+        UUID reportUuid = null;
+        if (headers.containsKey(REPORT_UUID_HEADER)) {
+            reportUuid = UUID.fromString(getNonNullHeader(headers, REPORT_UUID_HEADER));
+        }
```

Comment:
```suggestion
        Map<String, Double> voltageLevelsIdsRestricted;
        try {
            voltageLevelsIdsRestricted = headers.get(VOLTAGE_LEVELS_IDS_RESTRICTED) != null ?
                objectMapper.readValue((String) headers.get(VOLTAGE_LEVELS_IDS_RESTRICTED), new TypeReference<>() { }) : null;
        } catch (JsonProcessingException e) {
            throw new UncheckedIOException(e);
        }

        UUID parametersUuid = Optional.ofNullable((String) headers.get(PARAMETERS_UUID_HEADER)).map(UUID::fromString).orElse(null);
        UUID reportUuid = Optional.ofNullable((String) headers.get(REPORT_UUID_HEADER)).map(UUID::fromString).orElse(null);
```

In fact, I dont know why we need call containsKey instead of get directly the value in a map

Reply:
Indeed. Moreover those UUID may be null so `getNonNullHeader` shouldn't be used anyway.

Done.